coderedcms/models/page_models.py

```diff
@@ -1053,3 +1067,268 @@ def serve_preview(self, request, mode):
+
+    # Override body to provide simpler content
+    body = StreamField(CONTENT_STREAMBLOCKS, null=True, blank=True)
+
+    address = models.TextField(
+        blank=True,
+        verbose_name=_("Address")
+    )
+    latitude = models.FloatField(
+        blank=True,
+        null=True,
+        verbose_name=_("Latitude")
+    )
+    longitude = models.FloatField(
+        blank=True,
+        null=True,
+        verbose_name=_("Longitude")
+    )
+    auto_update_latlng = models.BooleanField(
+        default=True,
+        verbose_name=_("Auto Update Latitude and Longitude"),
+        help_text=_("If checked, automatically update the latitude and longitude when the address is updated.")
+    )
+    map_title = models.CharField(
+        blank=True,
+        max_length=255,
+        verbose_name=_("Map Title"),
+        help_text=_("If this is filled out, this is the title that will be used on the map.")
+    )
+    map_description = models.CharField(
+        blank=True,
+        max_length=255,
+        verbose_name=_("Map Description"),
+        help_text=_("If this is filled out, this is the description that will be used on the map.")
+    )
+    website = models.TextField(
+        blank=True,
+        verbose_name=_("Website")
+    )
+    phone_number = models.CharField(
+        blank=True,
+        max_length=255,
+        verbose_name=_("Phone Number")
+    )
+
+    content_panels = (
+        CoderedWebPage.content_panels[:1] + 
+        [
+            FieldPanel('address'),
+        ] +
+        CoderedWebPage.content_panels[1:] +
+        [
+            FieldPanel('website'),
+            FieldPanel('phone_number'),
+        ]
+    )
+
+    layout_panels = (
+        CoderedWebPage.layout_panels +
+        [
+            MultiFieldPanel(
+                [
+                    FieldPanel('map_title'),
+                    FieldPanel('map_description'),
+                ],
+                heading=_('Map Layout')
+            ),
+        ]
+    )
+
+    settings_panels = (
+        CoderedWebPage.settings_panels + 
+        [
+            MultiFieldPanel(
+                [
+                    FieldPanel('auto_update_latlng'),
+                    FieldPanel('latitude'),
+                    FieldPanel('longitude'),
+                ],
+                heading=_("Location Settings")
+            ),
+        ]
+    )
+
+    @property
+    def get_geojson_name(self):
+        return self.map_title or self.title
+
+    @property
+    def get_geojson_description(self):
+        return self.map_description
+
+    @property
+    def render_pin_description(self):
+        return mark_safe(
+            """
+                <b>{0}</b>
+                <p>{1}</p>
+                <p><a href='{2}''>View Location</a></p>
+            """
+            .format(
+                self.title,
+                self.address,
+                self.url
+            )
+        )
+
+    @property
+    def render_list_description(self):
```

Comment:
Not a fan of generating HTML inside a model. Is this method really needed? Couldn't this just be part of the html template?

Reply:
This gets called in the template as well as the maps.js file.  If we only had to render it in the template I would agree.  But doing it this way prevents people from needing to rewrite the javascript if they want to change the layout.

Reply:
Hmm... interesting. Could it get extracted into an {% include %} then, to get rendered in both places?